apps/web/src/api/hooks/integrations/use-integration-limit.ts

```diff
@@ -1,21 +1,25 @@
 import { useQuery } from '@tanstack/react-query';
 import { ChannelTypeEnum } from '@novu/shared';
 import { getIntegrationLimit } from '../../integration';
+import { useActiveIntegrations } from '.';
 
 export function useIntegrationLimit(type: ChannelTypeEnum) {
   const selfHosted = process.env.REACT_APP_DOCKER_HOSTED_ENV === 'true';
+  const { integrations = [] } = useActiveIntegrations();
 
   const {
     data = { limit: 0, count: 0 },
     isLoading,
     refetch,
   } = useQuery(['integrationLimit', type], () => getIntegrationLimit(type), {
-    enabled: selfHosted,
+    enabled: selfHosted && integrations.filter((integration) => integration.channel === type).length === 0,
```

Comment:
Worth extract this to a function or constant `isEnabled` as we reuse it in the file. 👍🏻